cpu/lpc11u34/periph/adc.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_lpc11u34
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Paul RATHGEB <paul.rathgeb@skynet.be>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include "cpu.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+
+/* guard in case that no ADC device is defined */
+#if ADC_NUMOF
+
+int adc_init(adc_t dev, adc_precision_t precision)
+{
+    adc_poweron(dev);
+
+    /* Set resolution */
+    switch (precision) {
+        case ADC_RES_6BIT:
+            LPC_ADC->CR |= (0x04 << 17);
+            break;
+        case ADC_RES_8BIT:
+            LPC_ADC->CR |= (0x02 << 17);
+            break;
+        case ADC_RES_10BIT:
+            LPC_ADC->CR |= (0x0 << 17);
+            break;
+        case ADC_RES_12BIT:
+        case ADC_RES_14BIT:
+        case ADC_RES_16BIT:
+            /* Resolution > 10 bits unsupported */
+            adc_poweroff(dev);
+            return -1;
+            break;
+    }
+
+    /* ADC frequency : 3MHz */
+    LPC_ADC->CR |= (15 << 8);
+
+    return 0;
+}
+
+int adc_sample(adc_t dev, int channel)
+{
+    if (channel > ADC_MAX_CHANNELS) {
+        return -1;
+    }
+    switch(channel) {
+         case 0:
+             /* Pin alternate mode : AD0 */
+             LPC_IOCON->TDI_PIO0_11 |= 2;
+             /* Pin ADMODE : Analog input */
+             LPC_IOCON->TDI_PIO0_11 &= ~(1 << 7);
```

Comment:
It could have been a good optimization. But unfortunately, as you can notice for the AD5 to AD7, the alternate function is not at 0x02 but at 0x01. 
I don't really like this part of code, but I don't know how to write it in a more esthetic way. I can maybe create mapping arrays or something like that, but in any case, I need to know the alternate function config for each pins.
A suggestion ?